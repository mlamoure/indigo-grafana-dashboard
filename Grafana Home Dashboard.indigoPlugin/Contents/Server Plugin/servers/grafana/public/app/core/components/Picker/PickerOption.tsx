import React, { Component } from 'react';

export interface Props {
  onSelect: any;
  onFocus: any;
  option: any;
  isFocused: any;
  className: any;
}

class UserPickerOption extends Component<Props, any> {
  constructor(props) {
    super(props);
    this.handleMouseDown = this.handleMouseDown.bind(this);
    this.handleMouseEnter = this.handleMouseEnter.bind(this);
    this.handleMouseMove = this.handleMouseMove.bind(this);
  }

  handleMouseDown(event) {
    event.preventDefault();
    event.stopPropagation();
    this.props.onSelect(this.props.option, event);
  }

  handleMouseEnter(event) {
    this.props.onFocus(this.props.option, event);
  }

  handleMouseMove(event) {
    if (this.props.isFocused) {
      return;
    }
    this.props.onFocus(this.props.option, event);
  }

  render() {
    const { option, children, className } = this.props;

    return (
      <button
        onMouseDown={this.handleMouseDown}
        onMouseEnter={this.handleMouseEnter}
        onMouseMove={this.handleMouseMove}
        title={option.title}
        className={`user-picker-option__button btn btn-link ${className}`}
      >
        <img src={option.avatarUrl} alt={option.label} className="user-picker-option__avatar" />
        {children}
      </button>
    );
  }
}

export default UserPickerOption;
