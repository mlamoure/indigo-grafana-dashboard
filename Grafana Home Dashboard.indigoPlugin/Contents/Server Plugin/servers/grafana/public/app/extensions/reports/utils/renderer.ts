// Get renderer version from its config field, return 1 by default
import { config } from '@grafana/runtime';

export const getRendererMajorVersion = (): number | null => {
  const version = config.rendererVersion;
  if (version === '') {
    return null;
  }

  const semverRegex =
    /^(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$/;
  const res = semverRegex.exec(version);

  let majorVersionStr = '';
  if (res && res.length > 1) {
    majorVersionStr = res[1];
  }

  if (majorVersionStr === '') {
    return 1;
  }

  let majorVersion = parseInt(majorVersionStr, 10);
  if (isNaN(majorVersion)) {
    majorVersion = 1;
  }

  return majorVersion;
};
