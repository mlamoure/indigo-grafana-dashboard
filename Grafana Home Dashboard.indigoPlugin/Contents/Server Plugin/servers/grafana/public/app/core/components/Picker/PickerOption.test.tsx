import React from 'react';
import renderer from 'react-test-renderer';
import PickerOption from './PickerOption';

const model = {
  onSelect: () => {},
  onFocus: () => {},
  isFocused: () => {},
  option: {
    title: 'Model title',
    avatarUrl: 'url/to/avatar',
    label: 'User picker label',
  },
  className: 'class-for-user-picker',
};

describe('PickerOption', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<PickerOption {...model} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
