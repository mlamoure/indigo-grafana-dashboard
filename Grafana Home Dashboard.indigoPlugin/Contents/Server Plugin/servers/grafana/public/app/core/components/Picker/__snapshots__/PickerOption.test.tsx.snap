// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PickerOption renders correctly 1`] = `
<button
  className="user-picker-option__button btn btn-link class-for-user-picker"
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseMove={[Function]}
  title="Model title"
>
  <img
    alt="User picker label"
    className="user-picker-option__avatar"
    src="url/to/avatar"
  />
</button>
`;
